lib/checks/resolvesToConstant.js

```diff
@@ -1,24 +1,21 @@
-const Node = require('../node');
+const {query} = require('math-nodes');
 
 // Returns true if the node is a constant or can eventually be resolved to
 // a constant.
 // e.g. 2, 2+4, (2+4)^2 would all return true. x + 4 would return false
 function resolvesToConstant(node) {
-  if (Node.Type.isOperator(node) || Node.Type.isFunction(node)) {
+  if (query.isApply(node)) {
     return node.args.every(
       (child) => resolvesToConstant(child));
   }
-  else if (Node.Type.isParenthesis(node)) {
-    return resolvesToConstant(node.content);
-  }
-  else if (Node.Type.isConstant(node, true)) {
+  else if (query.isNumber(node)) {
     return true;
   }
-  else if (Node.Type.isSymbol(node)) {
+  else if (query.isIdentifier(node)) {
     return false;
   }
-  else if (Node.Type.isUnaryMinus(node)) {
-    return resolvesToConstant(node.args[0]);
+  else if (node.type === 'Parentheses') {
```

Comment:
TBH, we should stop flattening b/c there's no reason to now.  `math-parser` produces an n-ary AST by default and `math-rules` returns n-ary ASTs.  Thus any nesting of nodes of the same type should be meaningful nesting.